src/components/SetFormBody.js

```diff
@@ -0,0 +1,165 @@
+import React, { PureComponent } from 'react'
+import { connect } from 'react-redux'
+import EditTask from './EditTask'
+import AddPopup from './Popup/AddPopup'
+import { editTask, addTask } from '../redux/reducers/todos/actions'
+
+const cleanForm = {
+  name: '',
+  description: '',
+  comment: '',
+  priority: '',
+}
+
+class SetFormBody extends PureComponent {
+  state = {
+    task: {
+      name: '',
+      description: '',
+      comment: '',
+      priority: '',
+    },
+  }
+
+  static getDerivedStateFromProps(
+    { showPopup, tasks, match },
+    state,
+  ) {
+    let task
+
+    if (showPopup === undefined) {
+      task = tasks.find(task => task.id === parseInt(match.params.id))
+
+      if (task && !state.task.timestamp)
+        return { ...state.task, task }
+    }
+
+    return null
+  }
+
+  getDate = () => {
+    const today = new Date()
+    return `${today.getDate()}.${today.getMonth() +
+      1}.${today.getFullYear()}`
+  }
+
+  getLastTaskId = tasks =>
+    tasks.length > 0 && tasks[tasks.length - 1].id
+
+  handleSubmit = event => {
```

Comment:
take a look on handle submit and try to refactor it. Now i think it is too complicated. It is possible to do it simpler.